### PR TITLE
WIP: Refactor map

### PR DESCRIPTION
**Pull request by lukashass**
Refactor map to use mapbox-gl. Planned steps:
* [x] render stops on map
* [x] make stops clickable
* [x] render buses on map
* [x] make buses clickable
* [x] set start coordinates to selected bus or stop when entering map
* [x] remember last coordinates when reentering map
* [x] show bus number in icon
* [x] show bus heading
* [x] bus focused arrow
* [x] create custom mapbox style
* [x] provide tiles from own tileserver

Bonus:
* [x] when focusing a bus highlight buses on same trip
* [ ] ~~display bus routes from OSM~~ moved to #24 
* [ ] ~~move buses along routes~~ moved to #24

### CODE REVIEW

File: spa/src/components/MapGL.vue [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  },
+  created() {
+    // We need to set mapbox-gl library here in order to use it in template
+    this.mapbox = Mapbox;
+    this.map = null;
+  },
+  mounted() {
+    this.$store.dispatch('map/load');
+
+    if (this.focusData) {
```
Comment (lukashass) [inline spa/src/components/MapGL.vue]: When the map has not been loaded before `this.focusData` will be `null` here even though it should be set.
How could this be fixed? @anbraten

Comment (lukashass) [reply spa/src/components/MapGL.vue]: The issue should be fixed by https://github.com/anbraten/opnv-live/pull/18/commits/7267e553c6bc8d727d57096e698cd5c359ecf729

**Comment by anbraten** [comment]
Please merge current master.

**Comment by anbraten** [comment]
ready to be merged?

**Comment by lukashass** [comment]
Yes, LGTM

**Pull request merged**